xarray/core/variable.py

```diff
@@ -2078,26 +2078,39 @@ def rolling_window(
         """
         if fill_value is dtypes.NA:  # np.nan is passed
             dtype, fill_value = dtypes.maybe_promote(self.dtype)
-            array = self.astype(dtype, copy=False).data
+            var = self.astype(dtype, copy=False)
         else:
             dtype = self.dtype
-            array = self.data
+            var = self
 
-        if isinstance(dim, list):
-            assert len(dim) == len(window)
-            assert len(dim) == len(window_dim)
-            assert len(dim) == len(center)
-        else:
+        if utils.is_scalar(dim):
+            for arg in [window, window_dim, center]:
+                assert utils.is_scalar(arg)
             dim = [dim]
             window = [window]
             window_dim = [window_dim]
             center = [center]
+        else:
+            assert len(dim) == len(window)
+            assert len(dim) == len(window_dim)
+            assert len(dim) == len(center)
```

Comment:
are these checks important? If they are, we might want to raise explicitly (`assert` is skipped with python's `-O` option)